tests/test_steps_edge.py

```
@@ -8,97 +15,107 @@
 
 
 def test_load_uploaded_file_missing_cols():
-    """Cover steps.py line 39."""
+    """Verify error message when uploaded file is missing required columns."""
     mock_file = MagicMock()
     mock_file.name = "test.csv"
     bad_df = pd.DataFrame({"Wrong": [1]})
     with patch("src.ui.steps.st") as mock_st:
-        mock_st.session_state.u_file = mock_file
         with patch("pandas.read_csv", return_value=bad_df):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)